spec/i18n_spec.rb

```diff
@@ -452,13 +454,18 @@ def extract_interpolation_arguments(translation)
       map(&:compact).map(&:first).to_set
   end
 
-  def flatten_hash(hash, parent_keys: [], out_hash: {}, &block)
+  def flatten_hash(hash, flatten_arrays: true, parent_keys: [], out_hash: {})
     hash.each do |key, value|
       if value.is_a?(Hash)
-        flatten_hash(value, parent_keys: parent_keys + [key], out_hash: out_hash, &block)
+        flatten_hash(value, flatten_arrays:, parent_keys: parent_keys + [key], out_hash:)
+      elsif value.is_a?(Array) && flatten_arrays
+        value.each_with_index do |item, idx|
+          flat_key = [*parent_keys, key, idx.to_s].join('.')
+          out_hash[flat_key] = item if item
+        end
       else
         flat_key = [*parent_keys, key].join('.')
-        out_hash[flat_key] = value
+        out_hash[flat_key] = value if value
```

Comment:
Are there keys with `nil` values?

Reply:
Yeah when I converted `month_names` arrays I made the 0th item `null`

https://github.com/18F/identity-idp/blob/a79bba606b9a7225f7931f64c9fea3a5dfe4ccfe/config/locales/en.yml#L421-L434

I guess I could go back and put an empty string there?

Reply:
Now that I think about it, it was null prior to the flattening too?

Reply:
hmm then it must have been a case of the old code not flattening arrays at all, so downstream code never encountered `nil` values

Reply:
Yeah, I suspect that's the case. I think keeping it null is probably preferable since it's been that way.